synapse/events/snapshot.py

```diff
@@ -67,49 +67,33 @@ class EventContext(object):
             Only set when state has not been fetched yet.
     """
 
-    __slots__ = [
-        "state_group",
-        "rejected",
-        "prev_group",
-        "delta_ids",
-        "prev_state_events",
-        "app_service",
-        "_current_state_ids",
-        "_prev_state_ids",
-        "_prev_state_id",
-        "_event_type",
-        "_event_state_key",
-        "_fetching_state_deferred",
-    ]
-
-    def __init__(self):
-        self.prev_state_events = []
-        self.rejected = False
-        self.app_service = None
+    state_group = attr.ib(default=None)
```

Comment:
just a note: you can annotate the types in this with `type=` -- for the optional ones you want to use `typing.Optional[int]` or whatever. mypy can read that info, and also some editor IDEs can as well, so it can be worth it :)

Reply:
mmm, good to know. I'll consider it next time ;)